src/hearings/converters/hearing-panel-other-panel-roles.converter.ts

```diff
@@ -0,0 +1,57 @@
+import { ActivatedRoute } from '@angular/router';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { State } from '../store';
+import { AnswerConverter } from './answer.converter';
+
+export class HearingPanelOtherPanelRolesConverter implements AnswerConverter {
+  constructor(protected readonly route: ActivatedRoute) {}
+
+  public transformAnswer(hearingState$: Observable<State>): Observable<string> {
+    return hearingState$.pipe(
+      map((state) => {
+        if (
+          !state.hearingRequest.hearingRequestMainModel.hearingDetails ||
+          !state.hearingRequest.hearingRequestMainModel.hearingDetails
+            .panelRequirements
+        )
+          return;
+
+        let selectedPanelRoles = '<ul>';
+        const panelDetailsResolverData =
+          this.route.snapshot.data.otherPanelRoles;
+        if (
+          state.hearingRequest.hearingRequestMainModel.hearingDetails
+            .panelRequirements.panelSpecialisms
+        ) {
+          state.hearingRequest.hearingRequestMainModel.hearingDetails.panelRequirements.panelSpecialisms.forEach(
+            (panelSpecialism) => {
+              const data = panelDetailsResolverData.find(
+                (panelRoles) => panelRoles.key === panelSpecialism
+              );
+              if (!data) {
+                let shouldSkip = false;
+                !shouldSkip &&
+                  panelDetailsResolverData.forEach((roles) => {
+                    const childNodesData =
+                      roles.child_nodes &&
+                      roles.child_nodes.find(
+                        (childNode) => childNode.key === panelSpecialism
+                      );
+                    if (!shouldSkip && childNodesData) {
+                      selectedPanelRoles += `<li>${roles.value_en} - ${childNodesData.value_en}</li>`;
+                      shouldSkip = true;
+                    }
+                    return;
```

Comment:
it probably best to recursively search the value from the key in this data object, rather than find to the depth of only secode level.